how-to/develop-with-buildd.rst

```diff
@@ -268,3 +292,88 @@ The buildd that you have just installed needs registering with Launchpad so that
     - Non-virtualized builders are much simpler: ``launchpad-buildd`` is cleaned synchronously over XML-RPC at the end of each build, and that's it.  If you use this, then you must be careful not to run any untrusted code on the builder (since a ``chroot`` or container escape could compromise the builder), and you'll need to uncheck "Require virtualized builders" on any PPAs, live file systems, recipes, etc. that you want to be allowed to build on this builder.
 
  4. After 30 seconds or so, the status of the builder on the builders page should be 'Idle'. This page does not auto-update, so refresh!
+
+Running a build on qastaging
+----------------------------
+
+We can use XML-RPC to interact also with qastaging/staging builders. 
+First of all, we should be able to ssh into our bastion and ssh into the ``launchpad-buildd-manager`` unit since it's the one that has the firewall rules to talk with builders.
+
+Then we should follow the same procedure to get the correct ``sha1sum`` of a backend image.
+
+.. code-block:: sh
+
+    $ ./manage-chroot -s bionic -a amd64 -i lxd get 
+    $ sha1sum livecd.ubuntu-base.lxd.tar.gz
+
+The command above will show us also the link from which the image is retrieved, let's save it since we will use it in our ``ensurepresent`` function.
+At this point we should select the builder that we want to interact with, we can navigate ``https://qastaging.launchpad.net/builders/qastaging-bos03-amd64-001`` and get the builder location.
+In this example ``http://qastaging-bos03-amd64-001.vbuilder.qastaging.bos03.scalingstack:8221``
+
+.. code-block:: python
+
+  import xmlrpclib
+  proxy = xmlrpclib.ServerProxy("http://qastaging-bos03-amd64-001.vbuilder.qastaging.bos03.scalingstack:8221/rpc")
+  proxy.status()
+
+  # Inject the backend image we retrieved before.
+  proxy.ensurepresent("<sha1sum from the previous step>", "<url from the previous step>", "admin", "admin")
+
+  # Start the build.
+  proxy.build('1-3', 'snap', '<sha1sum from the previous step>', {}, {'name': 'test-build', 'image_type': 'lxd', 'series': 'bionic', 'arch_tag': 'amd64', 'git_repository': 'https://github.com/tomwardill/test-docker-repo.git', 'archives': ['deb http://archive.ubuntu.com/ubuntu bionic main restricted', 'deb http://archive.ubuntu.com/ubuntu bionic-updates main restricted', 'deb http://archive.ubuntu.com/ubuntu bionic universe']})
+
+  # After that we can clean the builder.
+  proxy.clean()
```

Comment:
Directly? Or would we wait for the build to be finished?

Reply:
This comment has not been addressed yet.

Reply:
Oh sorry, I missed that. We should wait that the build finishes and then we can clean the build. I should update the comment here :)